agent/command_test.go

```diff
@@ -229,6 +229,58 @@ func (s *CommandSuite) setUpConfigAndProject(projYml string) {
 	s.tc.project = p
 	s.tc.taskConfig.Project = p
 }
+
+func (s *CommandSuite) TestUnsetFunctionVarsProjectFlag() {
+	s.tc.unsetFunctionVarsDisabled = true
+	projYml := `
+unset_function_vars: true 
+functions:
+  yes:
+    vars: 
+      key1: "functionVar"
+    command: shell.exec
+    params:
+        shell: bash
+        script: |
+          echo "hi"
+`
+
+	s.setUpConfigAndProject(projYml)
+
+	func1 := model.PluginCommandConf{
+		Function:    "yes",
+		DisplayName: "function",
+		Vars:        map[string]string{"key1": "functionVar"},
+	}
+
+	cmds := []model.PluginCommandConf{func1}
+	err := s.a.runCommandsInBlock(s.ctx, s.tc, cmds, runCommandsOptions{}, "")
+	s.NoError(err)
+
+	key1Value := s.tc.taskConfig.Expansions.Get("key1")
+	s.Equal("expansionVar", key1Value, "globalVar should be set back to what it was before the function ran")
+
+	s.tc.unsetFunctionVarsDisabled = true
```

Comment:
nit: the test above this seems unrelated to the test below this, so I would split it into two test cases.

Generally, keeping your test cases as specific, short, and precise as possible is preferable compared to a longer test since it reduces the possibility of having a bug in the test due to it doing a lot of things (especially where the later tests somehow depend on the state modifications of the previous tests).

Reply:
It's two sides of the same coin. It tests that it does work when set to true and doesn't work when set to false. Keeping it in one test will also make it easier when removing it, which we will do since it's temporary. This also reuses the function and commands definition.